File: 90.py
def permute(nums):
    res = []
    used = [False] * len(nums)
    curr = []

    def backtrack():
        if len(curr) == len(nums):
            res.append(curr.copy())
            return
        for i in range(len(nums)):
            if not used[i]:
                used[i] = True
                curr.append(nums[i])
                backtrack()
                curr.pop()
                used[i] = False

    backtrack()
    return res

print(permute([1,2]))

File: 90.js
function permute(nums) {
    const res = [];
    const used = Array(nums.length).fill(false);
    const curr = [];

    function backtrack() {
        if (curr.length === nums.length) {
            res.push([...curr]);
            return;
        }
        for (let i = 0; i < nums.length; i++) {
            if (!used[i]) {
                used[i] = true;
                curr.push(nums[i]);
                backtrack();
                curr.pop();
                used[i] = false;
            }
        }
    }

    backtrack();
    return res;
}

console.log(permute([1,2]));
